pageserver/src/tenant/secondary/downloader.rs

```diff
@@ -498,6 +551,24 @@ impl<'a> TenantDownloader<'a> {
 
         tracing::debug!("Wrote local heatmap to {}", heatmap_path);
 
+        // Clean up any local layers that aren't in the heatmap.  We do this first for all timelines, on the general
+        // principle that deletions should be done before writes whereever possible, and so that we can use this
+        // phase to initialize our SecondaryProgress.
+        {
+            let heatmap_stats = heatmap.get_stats();
+            // We will construct a progress object, and then populate its initial "downloaded" numbers
+            // while iterating through local layer state in [`Self::prepare_timelines`]
+            let mut progress_init = SecondaryProgress {
+                layers_total: heatmap_stats.layers,
+                bytes_total: heatmap_stats.bytes,
+                heatmap_mtime: Some(heatmap_mtime),
+                layers_downloaded: 0,
+                bytes_downloaded: 0,
+            };
```

Comment:
Yeah, I originally didn't do this because this function didn't have the heatmap_mtime needed to construct it, but it's cleaner to pass in the mtime than to use the &mut -- ee1555d91